### PR TITLE
Add logging for privacy-related actions

### PR DESCRIPTION
**Pull request by tomasr8**
Logs when
- Participant visibility settings are changed
- Participant changes their consent to update
- Participant is manually hidden/shown again

### CODE REVIEW

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 1]
```diff
@@ -600,8 +600,20 @@ def _process(self):
 class RHRegistrationHide(RHManageRegistrationBase):
     """Hide a registration from the participants lists."""
 
+    def _log_changes(self):
+        log_data = {'Email': self.registration.email}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py:604]: the friendly_id would be useful as well (either in the data or maybe even in the title of the log message)

File: indico/modules/events/registration/util.py [hunk 2]
```diff
@@ -413,6 +413,10 @@ def modify_registration(registration, data, management=False, notify_user=True):
             setattr(registration, key, value)
     if not management and regform.needs_publish_consent:
         registration.consent_to_publish = data.get('consent_to_publish', RegistrationVisibility.nobody)
+        registration.log(EventLogRealm.participants, LogKind.change,
+                         'Registration', f'Consent to publish modified: {registration.full_name}',
+                         session.user, data={'Email': registration.email})
```
Comment (ThiefMaster) [inline indico/modules/events/registration/util.py:418]: Including the old/new value would be a good idea ;)

File: indico/modules/events/registration/controllers/management/privacy.py [hunk 3]
```diff
@@ -25,6 +28,30 @@
...
+    def _get_changes(self, participant_visibility, public_visibility, visibility_duration, retention_period):
+        log_fields = {
+            'participant_visibility': {'title': 'Visibility to participants', 'type': 'string'},
+            'public_visibility': {'title': 'Visibility to everyone', 'type': 'string'},
+            'visibility_duration': {'title': 'Visibility duration', 'type': 'number'},
+            'retention_period': {'title': 'Retention period', 'type': 'number'},
+        }
+
+        changes = {
+            # Call str(..) because the enum title is a lazy string which is not JSON serializable
```
[the remark this answers is not in the document]
Comment (tomasr8) [reply indico/modules/events/registration/controllers/management/privacy.py]: Actually noticed that `make_diff_log` can do that automatically so no need for the conversion here at all

File: indico/modules/events/registration/util.py [hunk 4]
```diff
@@ -412,7 +413,14 @@ def modify_registration(registration, data, management=False, notify_user=True):
                 personal_data_changes[key] = value
             setattr(registration, key, value)
     if not management and regform.needs_publish_consent:
-        registration.consent_to_publish = data.get('consent_to_publish', RegistrationVisibility.nobody)
+        consent_to_publish = data.get('consent_to_publish', RegistrationVisibility.nobody)
+        changes = make_diff_log({'consent_to_publish': (registration.consent_to_publish, consent_to_publish)},
+                                {'consent_to_publish': {'title': 'Consent to publish'}})
+        registration.consent_to_publish = consent_to_publish
+        registration.log(EventLogRealm.participants, LogKind.change, 'Registration',
+                         f'Consent to publish modified: {registration.full_name}',
```
Comment (ThiefMaster) [inline indico/modules/events/registration/util.py:421]: It looks like you are logging this now even if nothing was changed (ie someone just editing their registration w/o touching it)...

**Pull request merged**